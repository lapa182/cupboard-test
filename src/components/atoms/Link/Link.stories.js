import React from "react";

import { storiesOf } from "@storybook/react";
import { withKnobs, text } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import Link from ".";

const link = text("Link", "#wine");
const withPreventDefault = handler => e => {
  e.preventDefault();
  handler(e);
};

storiesOf("Link", module)
  .addDecorator(withKnobs)
  .add("default", () => (
    <Link href={link} onClick={withPreventDefault(action("link-click"))}>
      Wine
    </Link>
  ));
