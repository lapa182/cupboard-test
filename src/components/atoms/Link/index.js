import * as React from "react";
import PropTypes from "prop-types";

const Link = ({ href, onClick, children }) => (
  <a href={href} onClick={onClick}>
    {children}
  </a>
);

Link.propTypes = {
  href: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export default Link;
