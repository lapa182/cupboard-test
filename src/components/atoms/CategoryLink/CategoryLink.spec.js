import React from "react";
import renderer from "react-test-renderer";
import Link from "./index";

describe("Link", () => {
  it("renders without crashing", () => {
    const linkEl = renderer
      .create(
        <Link id="001-01" href="#wine" onClick={() => {}}>
          Wine
        </Link>
      )
      .toJSON();
    expect(linkEl).toMatchSnapshot();
  });
});
