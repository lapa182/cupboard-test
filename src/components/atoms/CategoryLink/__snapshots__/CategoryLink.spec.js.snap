// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders without crashing 1`] = `
<a
  href="001-01"
  onClick={[Function]}
>
  Wine
</a>
`;
